--- test_util.py	2020-06-08 13:57:33.640469551 -0700
+++ test_util_new.py	2020-06-08 14:08:17.988496613 -0700
@@ -85,6 +85,7 @@
 from tensorflow.python.util.tf_export import tf_export
 from tensorflow.python.util.compat import collections_abc
 
+import tensorflow as tf
 
 # If the below import is made available through the BUILD rule, then this
 # function is overridden and will instead return True and cause Tensorflow
@@ -1788,6 +1789,7 @@
     # Avoiding calling setUp() for the poorly named test_session method.
     if self.id().endswith(".test_session"):
       self.skipTest("Not a test.")
+    tf.compat.v1.disable_eager_execution()
 
   def tearDown(self):
     for thread in self._threads:
